### PR TITLE
test: add nix dev shell env

### PR DESCRIPTION
**Pull request by HaxSam**
# Pull Request Template

## Type of change

- [ ] Bug fix
- [x] Feature
- [ ] Documentation update

## Description

Added nix dev shell env.
To use it go in the repo and use `nix-shell` in the dev shell you have all bin that ani-cli uses.
With arguments you can also add vlc, chromecast support and so on is commented in the `shell.nix` file

## Checklist

- [ ] any anime playing
- [ ] bumped version
---
- [ ] next, prev and replay work
- [ ] `-c` history and continue work
- [ ] `-d` downloads work
- [ ] `-s` syncplay works
- [ ] `-q` quality works
- [ ] `-v` vlc works
- [ ] `-e` select episode works
- [ ] `-S` select index works
- [ ] `-r` range selection works
- [ ] `--dub` and regular (sub) mode both work
- [ ] all providers return links (not necessarily on a single anime, use debug mode to confirm)
---
- [ ] `-h` help info is up to date
- [ ] Readme is up to date
- [ ] Man page is up to date

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Non-whole episodes: Tensei shitara slime datta ken (ep. 24.5, ep. 24.9)

### CODE REVIEW

**Comment by justchokingaround** [comment]
balls

File: shell.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ 
+  pkgs ? import <nixpkgs> {}, 
+  withMpv ? true,
+  withVlc ? false,
+  withIina ? false,
```
Comment (port19x) [inline shell.nix:5]: are you sure these are defined given just this file? Seems a bit odd

Comment (port19x) [reply shell.nix:5]: Or is that part of the upstream nixos-unstable package?

Comment (HaxSam) [reply shell.nix:5]: They are arguments that u can pass. Its supported in the nix-unstable package of ani-cli 4.6.
see here [ani-cli nixpkg](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/video/ani-cli/default.nix#L57)

**Comment by port19x** [comment]
Since this is supposed to be a dev shell, good extra dependencies to include would be `shfmt` and `shellcheck`, as those are part of the CI pipeline as well

**Comment by port19x** [comment]
If those are included I may use this myself, as shellcheck has way too many deps

**Comment by HaxSam** [comment]
No problem, I can include that

**Comment by justchokingaround** [comment]
also include the command to run the nix shell in question pls

**Comment by HaxSam** [comment]
> also include the command to run the nix shell in question pls

Where should I include it? It's already as comment in the `shell.nix`

**Comment by justchokingaround** [comment]
oh, my bad didn't see. ignore my comment

**Comment by port19x** [comment]
> > also include the command to run the nix shell in question pls
> 
> Where should I include it? It's already as comment in the `shell.nix`

This would fit well towards the bottom of contributing.md

**Comment by port19x** [comment]
Like under a `## Development with nix` you could outline the usage and link to an entrypoint to external docs.
I find that preferable to comments within the file itself

**Comment by Derisis13** [comment]
Does this mean we're merging packaging scripts?

**Comment by port19x** [comment]
> Does this mean we're merging packaging scripts?

No, nix is a heavily overloaded tool. What we're merging is a dev-environment wrapper.
Alternatives would include `devcontainer` and vanilla `Docker`

**Comment by Derisis13** [comment]
A dev-environment wrapper? So essentially a fakeroot environment if someone wants to work on the project but not install the dependencies?

**Comment by port19x** [comment]
> A dev-environment wrapper? So essentially a fakeroot environment if someone wants to work on the project but not install the dependencies?

yes, pretty much

**Comment by HaxSam** [comment]
Is all good now @port19x?

**Review by port19x — APPROVED** [review]

**Pull request merged**